vm/src/builtins/complex.rs

```diff
@@ -103,7 +103,7 @@ fn inner_div(v1: Complex64, v2: Complex64, vm: &VirtualMachine) -> PyResult<Comp
 
 fn inner_pow(v1: Complex64, v2: Complex64, vm: &VirtualMachine) -> PyResult<Complex64> {
     if v1.is_zero() {
-        return if v2.im != 0.0 {
+        return if v2.im != 0.0 || (v1.im.is_zero() && v2.re().is_negative()) { 
```

Comment:
The patch seems not to be based on the recent upstream.

#5143 has similar changes

Reply:
Yes, sorry about this. When i first fixed this error it was from [this issue](https://github.com/RustPython/RustPython/issues/5138) which wasn't closed.